### PR TITLE
MERGE: Talkingsystem/development lukas

### PR DESCRIPTION
**Pull request by Thabb**
Its at least some kind of a DialogueSystem. And before it gets even bigger...

### CODE REVIEW

File: Assets/Scripts/Dialogue/DialogueHandler.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        Cursor.lockState = CursorLockMode.None;
+
+        StartCoroutine(DialogueLoop());
+    }
+
+    private IEnumerator DialogueLoop()
+    {
+        foreach(DialogueObject dialogueObject in dialogueObjects)
+        {
+            bool a = true;
```
Comment (Mimonsi) [inline Assets/Scripts/Dialogue/DialogueHandler.cs]: What is this boolean for?

File: Assets/Scripts/Dialogue/DialogueHandler.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        DialogueOption = 1f;
+        dialogueInterface.transform.GetChild(0).GetChild(1).gameObject.SetActive(false);
+    }
+    public void decision3()
+    {
+        DialogueOption = 2f;
+        dialogueInterface.transform.GetChild(0).GetChild(1).gameObject.SetActive(false);
+    }
+}
```
Comment (Mimonsi) [inline Assets/Scripts/Dialogue/DialogueHandler.cs]: This redundand code could be made with a variable i = 1, 2, 3 etc.

**Review by Mimonsi — CHANGES_REQUESTED** [review]
Could be made a bit prettier :P Maybe it makes sense to implement it like this and overhaul it later. I found a few bugs when testing, for example the dialog doesn't work as intentended if opened for the second time.

File: Assets/Scripts/Dialogue/DialogueHandler.cs (hunk 1 above)
Comment (Thabb) [reply Assets/Scripts/Dialogue/DialogueHandler.cs]: nothing

**Comment by Thabb** [comment]
Resolved the bug and fixed the rest =)

File: Assets/Scripts/Dialogue/DialogueHandler.cs [hunk 3]
```diff
@@ -83,19 +83,9 @@ private void EndDialogue()
...
-    public void decision2()
-    {
-        DialogueOption = 1f;
-        dialogueInterface.transform.GetChild(0).GetChild(1).gameObject.SetActive(false);
-    }
-    public void decision3()
-    {
-        DialogueOption = 2f;
+        DialogueOption = i;
         dialogueInterface.transform.GetChild(0).GetChild(1).gameObject.SetActive(false);
```
Comment (Mimonsi) [inline Assets/Scripts/Dialogue/DialogueHandler.cs]: Perfect, exactly as I imagined

**Review by Mimonsi — APPROVED** [review]

**Pull request merged**